src/resources/startup.py

```diff
@@ -305,7 +305,7 @@ def Execute(self, request, context):
                 output_shape = output_np_array.shape
 
                 # We need to serialize TYPE_STRING
-                if output_np_array.dtype.type is np.object or output_np_array.dtype.type is np.bytes_:
+                if output_np_array.dtype == np.object or output_np_array.dtype.type is np.bytes_:
```

Comment:
Good catch. This also brought to my attention a relevant change that needs to be made in triton_python_backend_utils.py i.e. https://github.com/triton-inference-server/python_backend/blob/main/src/resources/triton_python_backend_utils.py#L71 and https://github.com/triton-inference-server/python_backend/blob/main/src/resources/triton_python_backend_utils.py#L55 where it should be `np.object` instead of `np.bytes_` since `np.bytes_` is not a dtype but a dtype.type. 

@Tabrizian the following lines need to be changed from `.dtype.type is np.object` to `.dtype == np.object` so you should open up a PR for the same
https://github.com/triton-inference-server/server/blob/master/qa/python_models/add_sub/model.py#L63 and https://github.com/triton-inference-server/server/blob/master/qa/python_models/sub_add/model.py#L63

Reply:
@CoderHam The decoder and encoder always create `np.bytes_` in these files: https://github.com/triton-inference-server/python_backend/blob/main/src/resources/triton_python_backend_utils.py#L71 and https://github.com/triton-inference-server/python_backend/blob/main/src/resources/triton_python_backend_utils.py#L55.

Check the deserializer that always creates np.bytes_: https://github.com/triton-inference-server/python_backend/blob/9e89c1018ef0a9cbd29c3c45ec0baffa7ccf0bc8/src/resources/startup.py#L130 (This is the same function from the Triton Python client)

I will create a separate PR to update the CI models.